pkg/controller/expectations.go

```diff
@@ -175,12 +175,18 @@ func (exp *ControlleeExpectations) isExpired() bool {
 	return clock.RealClock{}.Since(exp.timestamp) > ExpectationsTimeout
 }
 
+func getFormattedKeyFuncErrMessage(err error) error {
+	const keyFuncChangedFormat = "KeyFunc was changed, %v"
+
+	return fmt.Errorf(keyFuncChangedFormat, err)
+}
+
 // SetExpectations registers new expectations for the given controller. Forgets existing expectations.
 func (r *ControllerExpectations) SetExpectations(controllerKey string, add, del int) {
 	exp := &ControlleeExpectations{add: int64(add), del: int64(del), key: controllerKey, timestamp: clock.RealClock{}.Now()}
 	glog.V(4).Infof("Setting expectations %#v", exp)
 	if err := r.Add(exp); err != nil {
-		panic(fmt.Errorf("KeyFunc was changed, %v", err))
+		panic(getFormattedKeyFuncErrMessage(err))
```

Comment:
See above,  maybe it would be cleaner to just call `panicWithKeyFuncMsg(err)`  ?  (or any suitable name you find :-) )

Reply:
Sure :)

I thought about doing it but wasn't sure if it was right since maybe this error message could, in the future, be used for something other than panic (logging?). Anyways we can always split up this tiny function in the future and your idea makes current code cleaner in the mean time.